lib/resty/core/regex.lua

```diff
@@ -343,7 +343,7 @@ local function re_match_compile(regex, opts)
 end
 
 
-local function re_match_helper(subj, regex, opts, ctx, want_caps, res, nth)
+local function re_match_helper(subj, regex, opts, ctx, want_caps, want_both, res, nth)
```

Comment:
The `want_both` parameter name does not look very clear to me. Will you come up with a better name? Also, it's better to make it the last parameter instead of adding into the middle of the parameter list.